Data-Structures/src/main/java/stacksandqueues/Queue.java

```diff
@@ -0,0 +1,39 @@
+package stacksandqueues;
+
+public class Queue<T> {
+  Node<T> front;
+  Node<T> rear;
+  public Queue() {
+  }
+
+  public void enqueue(T value) {
+    Node<T> newNode = new Node(value, null);
+    if(rear == null) {
+      front = newNode;
+      rear = newNode;
+    } else {
+      rear.next = newNode;
+      rear = newNode;
+    }
+  }
+
+  public T dequeue() {
+    Node<T> temp;
+    if(front == null) {
```

Comment:
Since `peek()` has this same code, you could have called `peek()` here to make your code a little more DRY